packages/solana-rpc-provider/lib/SolanaRpcProvider.ts

```diff
@@ -0,0 +1,226 @@
+
+    return transactions.map((transaction) => this._normalizeTransaction(transaction))
+  }
+
+  async getBalance(addresses: (string | Address)[]): Promise<BigNumber> {
+    const promiseBalances = await Promise.all(
+      addresses.map(async (address) => {
+        try {
+          const _addr = typeof address === 'object' ? address.address : address
+
+          const publicKey = new PublicKey(_addr)
+
+          const balance = await this.connection.getBalance(publicKey)
+
+          return new BigNumber(balance)
+        } catch (err) {
+          if (err.message && err.message.includes('does not exist while viewing')) {
+            return new BigNumber(0)
+          }
+          throw err
+        }
+      })
+    )
+
+    return promiseBalances
+      .map((balance) => new BigNumber(balance))
+      .reduce((acc, balance) => acc.plus(balance), new BigNumber(0))
+  }
+
+  async sendSweepTransaction(address: string | Address, fee?: number): Promise<Transaction<any>> {
+    const sender = await this.getMethod('getUnusedAddress')()
+
+    const [balance, blockHash] = await Promise.all([
+      this.getBalance([sender.address]),
+      this.connection.getRecentBlockhash()
+    ])
+
+    const _fee = blockHash.feeCalculator.lamportsPerSignature
+
+    return await this.getMethod('sendTransaction')({
+      to: address,
+      value: balance.minus(_fee)
+    })
+  }
+
+  async sendRawTransaction(rawTransaction: string): Promise<string> {
+    const wireTransaciton = Buffer.from(rawTransaction)
+    return await this.connection.sendRawTransaction(wireTransaciton)
+  }
+
+  async getProgramAccounts(programId: PublicKey): Promise<
+    {
+      pubkey: PublicKey
+      account: AccountInfo<Buffer>
+    }[]
+  > {
+    return await this.connection.getProgramAccounts(new PublicKey(programId))
+  }
+
+  async getAccountInfo(address: string): Promise<AccountInfo<Buffer>> {
+    return await this.connection.getAccountInfo(new PublicKey(address))
+  }
+
+  async getMinimumBalanceForRentExemption(dataLength: number): Promise<number> {
+    return await this.connection.getMinimumBalanceForRentExemption(dataLength)
+  }
+
+  async sendAndConfirmTransaction(transaction: SolTransaction, accounts: any[]): Promise<string> {
+    return await sendAndConfirmTransaction(this.connection, transaction, accounts)
+  }
+
+  async getAddressHistory(address: string): Promise<ConfirmedSignatureInfo[]> {
+    return await this.connection.getConfirmedSignaturesForAddress2(new PublicKey(address))
+  }
+
+  async _includeTransactions(blockNumber: number): Promise<Transaction<any>[]> {
+    const confirmedSignatures = await this.connection.getConfirmedBlockSignatures(blockNumber)
+
+    const blockTransactions = confirmedSignatures.signatures.map((signature) => this.getTransactionByHash(signature))
+
+    return await Promise.all(blockTransactions)
+  }
+
+  async _deploy(signer: any): Promise<string> {
```

Comment:
The deploy should be part of sendTransaction. Because the consumer should be able to send any kind of transaction using sendTransaction. E.g. regular transfer, contract call, deployment. 
Maybe a good idea would be if there is a missing receiver when calling sendTransaction to invoke the deploy?

Also, the `Bytecode` should be passed as parameter, because right now you can only deploy the HTLC. The idea is that the CAL can be injected and used by 3rd party apps for any kind of interactions with the chain.

Let's discuss this offline